### ISSUE TITLE
Language Server for VSCode or other popular IDE's

### ISSUE THREAD

**Issue by tnekohue** [hlorenzi/customasm#105]
After working on a rather stupidly large project, I find myself wishing for autocompletion and syntax highlighting to make my life easier.
This is a suggestion/request to have someone make this possible.

**Comment by tnekohue** [comment]
Yeah, it would be a bit difficult, especially because of how complex the assembler is. I would use notepad++, but its not available under linux right now. This honestly seems complicated enough to warrant a separate project. Somehow the assembler knows how to match instructions, so it might be a good idea to start the autocompletion/highlighting there.

**Comment by hlorenzi** [comment]
I don't know if you're already using the [VSCode basic highlighting extension](https://marketplace.visualstudio.com/items?itemName=hlorenzi.customasm-vscode), but it does more or less what Notepad++ would do.

**Comment by parasyte** [comment]
I wrote a syntax definition for Sublime Text: https://github.com/parasyte/sublime-syntax-customasm It is fairly complete, but I'm not certain of its accuracy. (See also #139 which would provide a great reference for improving the accuracy of the syntax parser.)
